File: 2275_largest_combination_with_bitwise_and_greater_than_zero.cpp
class Solution {
public:
    int largestCombination(vector<int>& candidates) {
        int peak = 0;
        for (int altitude = 0; altitude < 32; ++altitude) {
            int beacon = 0;
            for (auto signal : candidates) {
                if (signal & (1 << altitude)) beacon++;
            }
            peak = max(peak, beacon);
        }
        return peak;
    }
};

File: 2275_largest_combination_with_bitwise_and_greater_than_zero.java
class Solution {
    public int largestCombination(int[] candidates) {
        int peak = 0;
        for (int layer = 0; layer < 32; layer++) {
            int blip = 0;
            for (int candidate : candidates) {
                if ((candidate & (1 << layer)) != 0) blip++;
            }
            peak = Math.max(peak, blip);
        }
        return peak;
    }
}
